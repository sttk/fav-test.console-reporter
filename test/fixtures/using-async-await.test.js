'use strict';

var chkEnv = require('../tool/chk-env');

var should = require('should');
var assert = require('assert');

function DB() {
}
DB.prototype.clear = function() {
  return new Promise(function (cb) {
    cb();
  });
};
DB.prototype.save = function() {
  return new Promise(function (cb) {
    cb();
  });
};
DB.prototype.find = function() {
  return new Promise(function(resolve) {
    resolve([tobi, loki, jane]);
  });
};

var tobi = 'Tobi',
    loki = 'Loki',
    jane = 'Jane';

var db = new DB();

if (!chkEnv.isSupportAsyncAwait()) {
  describe('#find()', function() {
    it('respond with matching records', function() {
    });
  });
  return;
}

eval("" +

"beforeEach(async function() {" +
"  await db.clear();" +
"  await db.save([tobi, loki, jane]);" +
"});" +
"" +
"describe('#find()', function() {" +
"  it('respond with matching records', async function() {" +
"    const users = await db.find({ type: 'User' });" +
"    users.should.have.length(3);" +
"  });" +
"});" +
"" +
"");
