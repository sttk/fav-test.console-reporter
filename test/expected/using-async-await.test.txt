

  #find()
    âœ“ respond with matching records


  1 passing (3ms)

